File: plugins/claude-md-plugin/core/tests/fixtures/go/contracts.go
package token

import (
	"errors"

	"github.com/golang-jwt/jwt/v5"
)

// Errors for token validation
var (
	ErrInvalidToken = errors.New("invalid token")
	ErrExpiredToken = errors.New("token expired")
)

// Claims represents the JWT claims.
type Claims struct {
	UserID string
	Role   string
}

// ValidateToken validates a JWT token.
// Precondition: token must be non-empty
// Postcondition: returns Claims or error
// Errors: ErrInvalidToken if token format is invalid
func ValidateToken(token string) (*Claims, error) {
	if token == "" {
		return nil, ErrInvalidToken
	}

	// ... validation logic using jwt package
	_ = jwt.New(jwt.SigningMethodHS256)

	return &Claims{UserID: "user123", Role: "admin"}, nil
}

// ProcessOrder processes an order.
// Precondition: order.ID must not be empty
// Precondition: order.Items must not be empty
// Postcondition: returns Receipt or error
func ProcessOrder(order Order) (*Receipt, error) {
	if order.ID == "" {
		return nil, errors.New("order ID required")
	}
	if len(order.Items) == 0 {
		return nil, errors.New("items required")
	}
	// ... process order
	return &Receipt{OrderID: order.ID, Total: 100}, nil
}

// Order represents an order.
type Order struct {
	ID    string
	Items []string
}

// Receipt represents a receipt.
type Receipt struct {
	OrderID string
	Total   int
}

// internalHelper is a private function.
func internalHelper(data string) string {
	return data
}

File: plugins/claude-md-plugin/core/tests/fixtures/python/contracts.py
"""Contract examples for Python code analyzer."""

from typing import Dict, Any
from .types import Claims


class InvalidTokenError(Exception):
    """Raised when a token is invalid."""
    pass


class ValidationError(Exception):
    """Raised when validation fails."""
    pass


def validate_token(token: str) -> Claims:
    """
    Validates a JWT token and returns the claims.

    Args:
        token: JWT token string (must be non-empty)

    Returns:
        Claims object with valid userId

    Raises:
        InvalidTokenError: if token format is invalid or expired
    """
    if not token:
        raise InvalidTokenError("Token is required")
    # ... validation logic
    return Claims(user_id="user123", role="admin")


def process_order(order: Dict[str, Any]) -> Dict[str, Any]:
    """
    Processes an order and returns a receipt.

    Args:
        order: Order dictionary with id and items

    Returns:
        Receipt dictionary with orderId and total

    Raises:
        ValidationError: if order is invalid
    """
    if not order.get("id"):
        raise ValidationError("Order ID required")
    if not order.get("items") or len(order["items"]) == 0:
        raise ValidationError("Items required")
    # ... process order
    return {"orderId": order["id"], "total": 100}


def _internal_helper(data: str) -> str:
    """Private helper function - should not be exported."""
    return data.strip()
